### ISSUE TITLE
Failed to terminate: Too many trys

### ISSUE THREAD

**Issue by fossil12** [nicklockwood/SwiftFormat#1458]
When I try to format the following Swift file I get an error:

```swift
class Example {
    enum Error: Swift.Error {
        case exampleError
    }
    
    init(string: String) throws {
        throw Error.exampleError
    }
    
    func exampleFunction() throws -> ManyInitExample {
        do {
            // let exampleString = try throwingExample()
            
            return ManyInitExample(
                a: try Example(string: try throwingExample()), // Comment this out
                // a: try Example(string: exampleString),
                b: try throwingExample(),
                c: try throwingExample(),
                d: try throwingExample(),
                e: try throwingExample(),
                f: try throwingExample(),
                g: try throwingExample(),
                h: try throwingExample(),
                i: try throwingExample()
            )
        } catch {
            throw error
        }
    }
    
    func throwingExample() throws -> String {
        throw Error.exampleError
    }
}

class ManyInitExample {
    init(a: Example, b: String, c: String, d: String, e: String, f: String, g: String, h: String, i: String) {
        
    }
}
```

Command:

```sh
swiftformat TooManyTrys.swift --swiftversion 5.7 --verbose
```

Log:

```shell
Running SwiftFormat...
Formatting [obmitted]/TooManyTrys.swift
-- error: Failed to terminate
error: 1 file could not be formatted.
```

SwiftFormat version:

```
0.51.10
```

Workaround:

1. Comment out the two commented in lines
2. Comment in the marked line

**Comment by facumenzella** [comment]
Can you try disabling `hoistTry` and see if that's still happening? You need to add `--disable indent` to your `.swiftformat` file

**Comment by nicklockwood** [comment]
@facumenzella I've found the issue. Basically because I simplified the rule to only clean up one `try` per pass, and SwiftFormat has a hard-coded limit of 10 passes before it gives up, if there are 10 `try`s in a single expression then it fails. I'll figure out a solution.

**Comment by nicklockwood** [comment]
@fossil12 fixed in 0.51.11

**Comment by fossil12** [comment]
@nicklockwood Thank you for the fix (including a new release) and this quick turnaround. 🙏🏼

I just tested it with my real codebase and it works now with no issues. 😃
